presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestPrestoContainerBasicQueries.java

```diff
@@ -0,0 +1,132 @@
+                .waitingFor(Wait.forLogMessage(".*======== SERVER STARTED ========.*", 1))
+                .withStartupTimeout(Duration.ofSeconds(Long.parseLong(CONTAINER_TIMEOUT)));
+
+        worker = new GenericContainer<>(PRESTO_WORKER_IMAGE)
+                .withExposedPorts(7777)
+                .withNetwork(network).withNetworkAliases("presto-worker")
+                .withFileSystemBind(BASE_DIR + "/testcontainers/nativeworker/velox-etc", "/opt/presto-server/etc", BindMode.READ_ONLY)
+                .withFileSystemBind(BASE_DIR + "/testcontainers/nativeworker/entrypoint.sh", "/opt/entrypoint.sh", BindMode.READ_ONLY)
+                .waitingFor(Wait.forLogMessage(".*Announcement succeeded: HTTP 202.*", 1));
+
+        coordinator.start();
+        worker.start();
+
+        // Wait for worker to announce itself.
+        TimeUnit.SECONDS.sleep(5);
+    }
+
+    @AfterClass
+    public void tearDown()
+    {
+        coordinator.stop();
+        worker.stop();
+    }
+
+    private Container.ExecResult executeQuery(String sql)
+            throws IOException, InterruptedException
+    {
+        // Command to run inside the coordinator container using the presto-cli.
+        String[] command = {
+                "/opt/presto-cli",
+                "--server",
+                "presto-coordinator:8081",
+                "--catalog",
+                "tpch",
+                "--schema",
+                "tiny",
+                "--execute",
+                sql
+        };
+
+        Container.ExecResult execResult = coordinator.execInContainer(command);
+        if (execResult.getExitCode() != 0) {
+            String errorDetails = "Stdout: " + execResult.getStdout() + "\nStderr: " + execResult.getStderr();
+            fail("Presto CLI exited with error code: " + execResult.getExitCode() + "\n" + errorDetails);
+        }
+        return execResult;
+    }
+
+    @Test
+    public void testBasics()
+            throws IOException, InterruptedException
+    {
+        String selectRuntimeNodes = "select * from system.runtime.nodes";
+        Container.ExecResult nodes = executeQuery(selectRuntimeNodes);
+        assertTrue(nodes.getStdout().contains("native-worker-1"), "Native worker is not present.");
+
+        String showCatalogs = "show catalogs";
+        Container.ExecResult catalogs = executeQuery(showCatalogs);
+        assertTrue(catalogs.getStdout().contains("tpch"), "tpch is not present.");
+
+        String showSession = "show session";
+        Container.ExecResult sessionProperties = executeQuery(showSession);
+        assertTrue(sessionProperties.getStdout().contains("native_aggregation_spill_all"), "native_aggregation_spill_all is not present.");
+    }
+
+    @Test
+    public void testFunctions()
+            throws IOException, InterruptedException
+    {
+        String countValues = "SELECT COUNT(*) FROM (VALUES 1, 0, 0, 2, 3, 3) as t(x)";
```

Comment:
These queries are evaluated at the co-ordinator only.

Might be good to add some queries that executes functions at the workers by using table columns.